backend/hdf5/PropertyHDF5.cpp

```diff
@@ -341,24 +416,74 @@ void PropertyHDF5::values(const std::vector<Variant> &values)
-        case DataType::UInt64: do_write_value<uint64_t>(dset, values); break;
+        case DataType::Bool:   do_write_value<bool>(dset, values);         break;
+        case DataType::Int32:  do_write_value<int32_t>(dset, values);      break;
+        case DataType::UInt32: do_write_value<uint32_t>(dset, values);     break;
+        case DataType::Int64:  do_write_value<int64_t>(dset, values);      break;
+        case DataType::UInt64: do_write_value<uint64_t>(dset, values);     break;
         case DataType::String: do_write_value<const char *>(dset, values); break;
-        case DataType::Double: do_write_value<double>(dset, values); break;
+        case DataType::Double: do_write_value<double>(dset, values);       break;
 #ifndef CHECK_SUPOORTED_VALUES
         default: assert(DATATYPE_SUPPORT_NOT_IMPLEMENTED);
 #endif
      }
 }
 
+Variant valueToVariant(const Value &val) {
+     switch (val.type()) {
+     case DataType::Bool:   return Variant(val.get<bool>());
+     case DataType::Int32:   return Variant(val.get<int32_t>());
+     case DataType::UInt32: return Variant(val.get<uint32_t>());
+     case DataType::Int64:  return Variant(val.get<int64_t>());
+     case DataType::UInt64: return Variant(val.get<uint64_t>());
+     case DataType::String: return Variant(val.get<std::string>());
+     case DataType::Double: return Variant(val.get<double>());
+#ifndef CHECK_SUPOORTED_VALUES
+     default: assert(DATATYPE_SUPPORT_NOT_IMPLEMENTED);
+#endif
+     }
+}
+
+std::vector<Value> PropertyHDF5::readOldstyleValues() const {
+    std::vector<Value> values;
+
+    DataSet dset = dataset();
+    DataType dtype = data_type_from_h5(dset.dataType());
+    NDSize shape = dset.size();
+
+    if (shape.size() < 1 || shape[0] < 1) {
+        return values;
+    }
+
+    assert(shape.size() == 1);
+    size_t nvalues = nix::check::fits_in_size_t(shape[0], "Can't resize: data to big for memory");
+
+    switch (dtype) {
+        case DataType::Bool:   do_read_old_value<bool>(dset, nvalues, values);     break;
+        case DataType::Int32:  do_read_old_value<int32_t>(dset, nvalues, values);  break;
+        case DataType::UInt32: do_read_old_value<uint32_t>(dset, nvalues, values); break;
+        case DataType::Int64:  do_read_old_value<int64_t>(dset, nvalues, values);  break;
+        case DataType::UInt64: do_read_old_value<uint64_t>(dset, nvalues, values); break;
+        case DataType::String: do_read_old_value<char *>(dset, nvalues, values);   break;
+        case DataType::Double: do_read_old_value<double>(dset, nvalues, values);   break;
+#ifndef CHECK_SUPOORTED_VALUES
+        default: assert(DATATYPE_SUPPORT_NOT_IMPLEMENTED);
+#endif
+    }
+    return values;
+}
+
 
 std::vector<Variant> PropertyHDF5::values(void) const
 {
     std::vector<Variant> values;
-
+    nix::FormatVersion ver(this->entity_file->version());
+    if ( ver < nix::FormatVersion({1, 1, 1})) {
+        for (Value v : readOldstyleValues()) {
```

Comment:
So, we read all the old stuff, but then throw it away? Maybe read directly into a `Variant`?

Reply:
hm, true, but then I need to have a second version for the uncertainty